### PR TITLE
[stable/redis-ha] Liveness probe just check instance liveness

### PR DESCRIPTION
**Pull request by Globegitter**
#### What this PR does / why we need it:
 @ssalaues We are running into the same issue as mentioned in https://github.com/helm/charts/pull/10178, i.e. that once one instance is down we sometimes can see the sentinels being stuck in an infinite loop of getting restarted by the livenessprobe an no way of recovering automatically. I think this is important to get unblocked for now that I made a PR to just change that. Would be great to get a fix in for that quickly.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped

### CODE REVIEW

**Comment by ssalaues** [comment]
@Globegitter Thanks for the PR! I agree this is important and should be addressed immediately. I just responded to the comment in #10178 . I'm trying to reproduce the issue you are both facing so I can validate the fixes as well.

**Comment by ssalaues** [comment]
@Globegitter I just posted a possible fix as well. I think adding a `timeout -t 3` or similar to the init ConfigMap might fix the endless sentinel crash loop on upgrade. Would you be able to try this out?
https://github.com/helm/charts/blob/master/stable/redis-ha/templates/redis-ha-configmap.yaml#L42

I really would prefer to avoid disabling the quorum checks as it's a huge step up from just a simple "ping" that doesn't say much about the state of the server.

**Comment by Globegitter** [comment]
@ssalaues Thanks for looking into this and yes I also think this should be solved properly and this was just the quickest way I could think of. But my question, if there is really a problem with the quorom, e.g. two sentinels of three are down for some reason so then the liveness check of the third sentinel would fail and k8s would try and restart. But what would that actually achieve? In which case would the restart fix anything? I am not just not a 100% sure about the idea / reasoning behind it.

**Comment by ssalaues** [comment]
@Globegitter Yeah I agree with you. The original reasoning was that a sentinel out of quorum would crash loop which a production environment would likely trigger an alert via Prometheus or similar. Basically a very obvious signal that something is wrong.

But you are correct in that nothing actually changes upon restart so it doesn't do anything beyond the above. It's likely that the quorum check makes more sense as a readiness probe but for now we should just move forward with your PR.

File: stable/redis-ha/templates/redis-ha-statefulset.yaml [hunk 1]
```diff
@@ -80,7 +80,7 @@ spec:
 {{- end }}
         livenessProbe:
           exec:
-            command: [ "sh", "/probes/check-quorum.sh", "{{ .Values.redis.port }}"]
+            command: [ "sh", "/probes/readiness.sh", "{{ .Values.redis.port }}"]
```
Comment (ssalaues) [inline stable/redis-ha/templates/redis-ha-statefulset.yaml:87]: Maybe a good idea to disable it on the sentinel probe as well.

**Comment by cpanato** [comment]
@Globegitter please rebase and bump the chart and check the comment made by @ssalaues thanks for your contribution!

File: stable/redis-ha/templates/redis-ha-statefulset.yaml (hunk 1 above)
Comment (Globegitter) [reply stable/redis-ha/templates/redis-ha-statefulset.yaml:87]: Done

**Comment by Globegitter** [comment]
@cpanato Done :)

**Comment by Globegitter** [comment]
@ssalaues Just merged latest master in again, would be great to get this merged if everything is looking good now :)

**Pull request merged**